### PR TITLE
Arena work

### PR DESCRIPTION
**Pull request by gozer**

### CODE REVIEW

File: nubis/puppet/consul.pp [hunk 1]
```diff
@@ -108,3 +124,5 @@
     environment => [ 'PATH="/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"', 'SHELL=/bin/bash' ],
     require     => File['/usr/local/sbin/consul-backup']
 }
+
+
```
Comment (tinnightcap) [inline nubis/puppet/consul.pp]: Two newlines. Is this intentional?

File: nubis/puppet/files/consul-dns-update [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -l
+
+# /usr/local/bin isn't set yet in our PATH
+export PATH=/usr/local/bin:$PATH
+
+# Only 1 member of the cluster needs to run this
+# if you are not the leader then just forget about it
+INSTANCE_ID=$(curl -s -fq http://169.254.169.254/latest/meta-data/instance-id)
```
Comment (tinnightcap) [inline nubis/puppet/files/consul-dns-update]: Curious why this is not in nubis-metadata?

File: nubis/terraform/main.tf [hunk 3]
```diff
@@ -1,65 +1,170 @@
...
+  region  = "${var.aws_region}"
+  version = "~> 0.1"
+}
+
+provider "random" {
+  version = "~> 0.1"
+}
+
+module "image" {
+  source = "github.com/gozer/nubis-deploy//modules/images?ref=feature%2Farena"
```
Comment (tinnightcap) [inline nubis/terraform/main.tf]: should this be pointing to gozer instead of nubisproject?

File: nubis/terraform/main.tf [hunk 4]
```diff
@@ -68,42 +173,62 @@ resource "aws_autoscaling_group" "consul" {
...
+
+  tag {
+    key                 = "ConsulClusterName"
+    value               = "consul-server-${element(var.arenas, count.index)}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "spot-enabled"
+    value               = "true"
```
Comment (tinnightcap) [inline nubis/terraform/main.tf]: should this be false for now?

File: nubis/terraform/main.tf [hunk 5]
```diff
@@ -473,26 +643,60 @@ resource "tls_self_signed_cert" "consul_web_public" {
...
+  certificate_body = "${element(tls_self_signed_cert.consul_web_ui.*.cert_pem, count.index)}"
+  private_key      = "${tls_private_key.consul_web.private_key_pem}"
+
+  # Amazon lies about key creation and availability
+
+  #provisioner "local-exec" {
+
+  #  command = "sleep 10"
+
+  #}
```
Comment (tinnightcap) [inline nubis/terraform/main.tf]: These are commented out, can they be removed?

File: nubis/puppet/files/consul-dns-update (hunk 2 above)
Comment (gozer) [reply nubis/puppet/files/consul-dns-update]: We don't have a tool for that yet.

**Pull request merged**